hashes/src/util.rs

```diff
@@ -243,7 +235,7 @@ mod test {
     fn borrow_slice_impl_to_vec() {
         // Test that the borrow_slice_impl macro gives to_vec.
         let hash = sha256::Hash::hash(&[3, 50]);
-        assert_eq!(hash.to_vec().len(), sha256::Hash::LEN);
+        assert_eq!(hash.as_ref().to_vec().len(), sha256::Hash::LEN);
```

Comment:
The test was testing presence of `to_vec` on the hash. Which I think is very questionable, so I suggest change it to testing that `as_ref` exists and works as expected. Although I think we should have both `AsRef<[u8]>` and AsRef<[u8; LEN]>`.

Reply:
What is the benefit of having both @Kixunil? I've had two goes #1462 now and its a bit curly so I just wanted to check in with you. Why is a reference to an array useful _as well as_ a slice, isn't a reference to an array functionally the same as a slice?

Reply:
Some algorithms may take `T: AsRef<[u8]>`, others `T: AsRef<[u8; N]>` depending on what those algorithms do. Having more of these standard traits can be very helpful.

Reply:
Ah yeah, cool. Thanks